libs/platform/ebpf_hash_table.c

```diff
@@ -3,52 +3,321 @@
 
 #include "ebpf_platform.h"
 
+// Buckets contain an array of pointers to data blocks containing key + value.
+// Buckets are immutable once inserted in to the hash-table and replaced when
+// modified.
+
+// Layout is:
+// ebpf_hash_table_t.buckets->ebpf_hash_bucket_header_t.entries->data
+// Keys are stored contiguously in ebpf_hash_bucket_header_t for fast
+// searching, data is stored separately to prevent RCU from causing loss of
+// updates.
+
+typedef struct _ebpf_hash_bucket_entry
+{
+    uint8_t* data;
+    uint8_t key[1];
+} ebpf_hash_bucket_entry_t;
+
+typedef struct _ebpf_hash_bucket_header
+{
+    size_t count;
+    ebpf_hash_bucket_entry_t entries[1];
```

Comment:
```suggestion
    _Field_size_(count)
    ebpf_hash_bucket_entry_t entries[1];
```
(see https://docs.microsoft.com/en-us/cpp/code-quality/annotating-structs-and-classes?view=msvc-160 for explanation of this suggestion)